Fix Tasks and Notes nodes continue to persist in the space navigation even after removing both applications from space settings - EXO-62454 Meeds-io/meeds#783

sofyenne opened the pull request:
Prior to this change when notes or tasks  applications was added to space and both application were removed from the space settings,the Notes and Tasks node continued to persist on the space navigation .The problem was that we couldn't  find the node to be removed  by the application name if the application name and saved node name were not the same  , for example for the tasks application the saved node name was `tasks` and the application name was `TasksManagement` .
After this change we will attempt to find the node to remove by the node pageRef name and the application id .

aycherif committed:
TASK-62301: Create FB maintenance and update projects versions/dependencies

IlhemEssaadi committed:
fix: fix space and user popover position - EXO-62433 - Meeds-io/meeds#706

sofyenne committed:
Fix Tasks and Notes nodes continue to persist in the space navigation even after removing both applications from space settings - EXO-62454 Meeds-io/meeds#783

ahamdi commented on component/core/src/main/java/org/exoplatform/social/core/space/impl/DefaultSpaceApplicationHandler.java:
```diff
@@ -420,7 +420,15 @@ protected void removeApplicationClassic(Space space, String appId, String appNam
+      if (removedNode == null){
+        //Try to get the node by the pageRef name and the appId
+        for (UserNode node : spaceUserNode.getChildren()){
+          if (node.getPageRef().getName().equals(appId)){
```
```suggestion
          if (appId.equals(node.getPageRef().getName())){

```

sofyenne replied:
fixed !

sofyenne committed:
clean code

GouadriaHanen committed:
fix: Can't not add a new phone number just after removing the last old one - EXO-62885 (#2389)

prior to this change, after adding all possible Phone or IMS contact and then removing them all, the add button is no more displayed since the property has no more children to display it as a multi-field so it is displayed as an input field
after this change, removing only the value of children from the property or all the children if it is new
makes the property always as multi-field

sofyenne committed:
Merge branch 'feature/maintenance' into Task-62454

sofyenne committed:
Merge branch 'feature/maintenance' into Task-62454
